apps/server/src/shared/domain/entity/course.entity.ts

```diff
@@ -95,14 +95,31 @@ export class Course
 	}
 
 	public getStudentIds(): EntityId[] {
-		if (!this.students) {
-			throw new InternalServerErrorException('Course.students is undefined. The course need to be populated.');
+		const studentIds = this.extractIds(this.students);
+		return studentIds;
+	}
+
+	public getTeacherIds(): EntityId[] {
```

Comment:
BTW this could also be getters like `get teacherIds()`. Would be good if we stay consistent in the long run...